stage3/react/modules/module05/README.md

```diff
@@ -0,0 +1,44 @@
+# Неделя 5
+
+## React. Custom app state
+
+
+
+
+!! Задание выполняется в личном приватном репозитории !!
+Нужно выполнять задание на базе предыдущего
+
+
+
+
+Что должно быть сделано:
+
+
+
+1) Создать отдельную ветку для этого задания
+2) Создать стейт менеджмент используя useReducer и Context API, где сохранять результаты поиска и данные формы. Теперь при переключении между страницами все данные должны сохраняться.
+3) Добавить в глобальный стейт
+    - **переключатели для сортировки**(как минимум три варианта сортировки). Выбираем параметр -> отправляем запрос к API с введенным параметром -> список результатов поиска обновляется
+    -  **переключатели для пагинации.** Должна быть возможность выбрать количество результатов на странице и номер страницы, на которой мы находимся сейчас + общее количество страниц. Выбираем параметр -> отправляем запрос к API с введенным параметром -> список результатов поиска обновляется
+4) Кастомер изменил требования к выводу деталей. Теперь при нажатии на каждый айтем из результатов поиска не нужно открывать модалку, а открывать информацию на отдельной странице. Также на этой странице добавить ссылку "Назад". В хедере отобразить текущее положение. Страница с деталями должна брать данные из глобального стейта.
+### Оценка
+
+
+
+
+Задание будет оцениваться ментором в марте. Для удобства проверки создайте Pull Request(**МЕРЖИТЬ НЕ НАДО**).
+
+
+
+
+1) Custom State Management - **9 баллов**
+2) Сортировка и пагинация  - **3 балла**
+3) Детали поиска в отдельной странице - **3 балла**
+
+
+
+Если хоть один из пунктов не выполнен, то задание оценивается в **0 баллов**.
+
+### Теория
+https://reactjs.org/docs/hooks-reference.html#usereducer
```

Comment:
useReducer: https://reactjs.org/docs/hooks-reference.html#usereducer
useContext: https://reactjs.org/docs/hooks-reference.html#usecontext

- https://dev.to/madv/usecontext-with-typescript-23ln
- https://dev.to/craigaholliday/using-the-usereducer-hook-in-react-with-typescript-27m1